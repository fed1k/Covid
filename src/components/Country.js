import React, { useEffect, useState } from 'react';

const Country = () => {
  const [myData, setMyData] = useState([]);
  const fetchCountry = async () => {
    const response = await fetch(`https://api.covid19tracking.narrativa.com/api/2020-03-10/country${window.location.pathname}`);
    const data = await response.json();
    setMyData(Object.entries(data.dates['2020-03-10'].countries)[0][1]);
  };
  useEffect(() => {
    fetchCountry();
  }, []);
  console.log(myData);
  return (
    <div className="country">
      {myData ? (
        <div className="country-item">
          <h1>{myData.name}</h1>
          <span>
            {myData.date}
            {' '}
            confirmed cases
            {' '}
            {myData.today_confirmed}
          </span>
          {myData.regions ? myData.regions.map((i) => (
            <div key={i.id} className="regional-data-items">
              <h4 key={i.name}>{i.name}</h4>
              <span>{i.today_confirmed}</span>
            </div>
          ))
            : <h5>No Data Available</h5>}
        </div>
      ) : <h1>Loading...</h1>}
    </div>
  );
};

export default Country;
