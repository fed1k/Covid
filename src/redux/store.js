import { configureStore } from '@reduxjs/toolkit';
import myReducer from './reducer';

const store = configureStore({
  reducer: {
    country: myReducer,
  },
});

export default store;
