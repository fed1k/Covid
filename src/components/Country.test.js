// import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Country from './Country';

test('should render correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Country />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
