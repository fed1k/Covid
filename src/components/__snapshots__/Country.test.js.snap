// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['should render correctly 1'] = `
<div
  className="country"
>
  <div
    className="country-item"
  >
    <img
      alt="FLAG IS NOT AVAILABLE"
      src="https://countryflagsapi.com/png/undefined"
    />
    <h1 />
    <span>
       
      confirmed cases
       
    </span>
    <h5>
      Regional data is not Available
    </h5>
  </div>
</div>
`;
